### PR TITLE
Added location support to Twitter and Facebook

### PR DESCRIPTION
**Pull request by aphex3k**
Hi,

I'd like to use Xamarin.Social for sharing pictures from a camera component in an app. I noticed that it yet lacks location support. I added my humble idea of location support and would like some input. 
- I was only able to test it on iOS (6 & 7) for lack of an actual Android test device and license.
- I would like to add it to App.net as well but am not yet ready to invest $100 for the developer account just yet.

The way it currently works is, that the app determines it's location and assigns lat and long values to the Location property of the item. If the values are set, we pass them verbatim to the Twitter Api. For Facebook we have to find a place_id first. An additional request is necessary for this. I imagine the API for App.net being similar to Twitter in this regard.

michael

### CODE REVIEW

File: src/Xamarin.Social/OAuth1Service.cs [hunk 1]
```diff
@@ -33,10 +33,6 @@ public abstract class OAuth1Service : Service
 		public OAuth1Service (string serviceId, string title)
 			: base (serviceId, title)
 		{
-			//
-			// This is a reliable URL to redirect to
-			//
-			CallbackUrl = new Uri ("http://www.facebook.com/connect/login_success.html");
```
Comment (ermau) [inline src/Xamarin.Social/OAuth1Service.cs]: This change isn't related to adding location support, please submit separately.

File: src/Xamarin.Social/Item.cs [hunk 2]
```diff
@@ -124,6 +124,22 @@ protected virtual void Dispose (bool disposing)
 		/// </summary>
 		public IList<FileData> Files { get; set; }
 
+		private Location _location = null;
```
Comment (ermau) [inline src/Xamarin.Social/Item.cs]: Don't use prefixes for fields.

File: Xamarin.Auth [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit b43fea8af787141c76c7e5ceb76ff2952dd92fd4
+Subproject commit 98a0594a01c043f7b49d31d204fb928c1f0db71f
```
Comment (ermau) [inline Xamarin.Auth]: Don't change the submodule commit unless you're updating to a newer revision on xamarin/master.

File: src/Xamarin.Social/Services/FacebookService.cs [hunk 4]
```diff
@@ -75,8 +77,40 @@ static string GetValueFromJson (string json, string key)
...
+
+			if (item.Location.Latitude != 0 && item.Location.Longitude != 0)
+			{
+				Request placeRequest = CreateRequest("GET", new Uri ("https://graph.facebook.com/search"), account);
+
+				placeRequest.Parameters ["type"] = "place";
+				placeRequest.Parameters ["center"] = item.Location.Latitude.ToString() + "," + item.Location.Longitude.ToString();
+				placeRequest.Parameters ["distance"] = "100";
+
+				await placeRequest.GetResponseAsync(cancellationToken).ContinueWith(reqTask => {
```
Comment (ermau) [inline src/Xamarin.Social/Services/FacebookService.cs]: Using ContinueWith along with `await` defeats the purpose of `await`, choose one. Additionally, use `.ConfigureAwait(false) on awaits inside libraries.

File: src/Xamarin.Social/Services/FacebookService.cs [hunk 5]
```diff
@@ -75,8 +77,40 @@ static string GetValueFromJson (string json, string key)
...
+
+					return ShareItemWithLocationAsync(item, account, placeId, cancellationToken);
+				});
+				;
+			} 
+			else
+			{
+				await ShareItemWithLocationAsync(item, account, null, cancellationToken).ContinueWith(reqTask => {
+
+					return reqTask;
```
Comment (ermau) [inline src/Xamarin.Social/Services/FacebookService.cs]: This is redundant, also needs `.ConfigureAwait(false)`.

File: src/Xamarin.Social/Services/FacebookService.cs [hunk 6]
```diff
@@ -75,8 +77,40 @@ static string GetValueFromJson (string json, string key)
...
+				});
+			}
+		}
+
+		public Task ShareItemWithLocationAsync (Item item, Account account, string placeId, CancellationToken cancellationToken)
```
Comment (ermau) [inline src/Xamarin.Social/Services/FacebookService.cs:119]: Why is this a separate public method if you're detecting whether it needs to be called in ShareItemAsync anyway?

File: src/Xamarin.Social/OAuth1Service.cs (hunk 1 above)
Comment (aphex3k) [inline src/Xamarin.Social/OAuth1Service.cs]: sure, no problem

File: src/Xamarin.Social/Item.cs (hunk 2 above)
Comment (aphex3k) [inline src/Xamarin.Social/Item.cs]: Can you rephrase that? I'm not sure I understand. I know _location is the name of a field here but what prefix are you referring to?

Comment (ermau) [inline src/Xamarin.Social/Item.cs]: The `_`
